openlibrary/plugins/ol_infobase.py

```diff
@@ -479,7 +479,7 @@ def process_edition_doc(self, doc):
         return doc
 
     def normalize_edition_title(self, title):
-        if isinstance(title, str):
+        if isinstance(title, bytes):
             title = title.decode('utf-8', 'ignore')
```

Comment:
This seems suspect to me in terms of Python3 compatibility, why don't we know whether title is bytes or str at this point?

Whatever `doc` is should be a collection of strings, and if it is retrieved over a network as bytes, it should be immediately decoded as utf8 as soon as possible, not decoded conditionally field by field much later.  @cclaus

Reply:
Yes.  @aasifkhan7 has an [array of pull requests](https://github.com/internetarchive/openlibrary/pulls/aasifkhan7) that replace urllib with the Requests library and look at the bytes vs. str issues on our network requests.  These PRs require careful review and should not all land at the same time.